iam_role.go

```diff
@@ -114,14 +118,26 @@ func NewIamRoleRequest(options *CreateIamRoleOptions) (*IamRoleRequest, error) {
 		return nil, fmt.Errorf("RoleName option must not be nil")
 	}
 
-	if options.RoleType == nil {
-		return nil, fmt.Errorf("RoleType option must not be nil")
+	if options.RoleType == nil && options.TrustPolicy == nil {
+		return nil, fmt.Errorf("Either RoleType or TrustPolicy option must not be nil")
+	}
+
+	if options.RoleType != nil && options.TrustPolicy != nil {
+		return nil, fmt.Errorf("RoleType and TrustPolicy options cannot be specified at the same time")
```

Comment:
 this can be done in fewer expressions, though it doesn't add much value
```suggestion
	trustPolicyExists := options.TrustPolicy == nil
	roleTypeExists := options.RoleType == nil
	if trustPolicyExists == roleTypeExists {
		return nil, fmt.Errorf("Either RoleType or TrustPolicy must be included, but not both")
	}
```